.github/workflows/integration_tests.yml

```diff
@@ -37,9 +37,17 @@ jobs:
     runs-on: ${{ needs.start-runner.outputs.label }}
     timeout-minutes: 600
     steps:
+      - name: Determine branch ref
+        id: branch_ref
+        run: |
+          if [ "${{ github.event_name }}" = "pull_request" ]; then
+            echo "branch_ref=refs/heads/${{ github.head_ref }}" >> $GITHUB_OUTPUT
+          else
+            echo "branch_ref=${{ github.ref }}" >> $GITHUB_OUTPUT
+          fi
       - uses: actions/checkout@v4
         with:
-          ref: ${{ github.ref }}
+          ref: "${{ steps.branch_ref.outputs.branch_ref }}"
```

Comment:
The output variable `branch_ref` created in the `branch_ref` step is used here for checking out the branch.